site/docs/v1/tech/apis/hosted-backend.adoc

```diff
@@ -0,0 +1,396 @@
+The top-level client Id for your application.
+
+[field]#redirect_uri# [type]#[String]# [optional]#Optional#::
+The URI that the browser will be redirected to at the end of the login flow. If not provided this will be the first of the Authorized Redirect URLs for your application.
+
+[field]#state# [type]#[String]# [optional]#Optional#::
+The value of this parameter will be echoed back in the [field]#state# parameter of the redirect URL at the end of the login flow.
+
+[field]#scope# [type]#[String]# [optional]#Optional# [default]#defaults to `openid offline_access`#::
+The OAuth 2 `scope` parameter to be passed to the authorize endpoint.
++
+Available scopes:
++
+* `openid` - This scope is used to request the `app.idt` Id token cookie be returned in the response
+* `offine_access` - This scope is used to request the `app.rt` refresh token cookie be returned in the response
+
+.Example Request URL
+[source,url]
+----
+https://auth.yourdomain.io/app/login/297ca84b-69a9-4508-8649-97644e1d0b3d?redirect_uri=https%3A%2F%2Fapp.yourdomain.io%2Fcallback&state=yourStateData&scope=offline_access
+----
+
+=== Response
+
+Successful invocations of this route will return a  `302` redirect to `/oauth2/authorize`. Other status codes indicate an error.
+
+[cols="1a,9a"]
+.Response Codes
+|===
+|Code |Description
+
+|200
+|This is likely an error case. The route will serve up an error page with html and details on what went wrong.
+
+|302
+|A successful request will redirect the user to `/oauth2/authorize` to log in.
+
+|403
+|A forbidden response typically means that the Origin of this request did not pass the FusionAuth CORS filter. You likely need to add your app to your link:/docs/v1/tech/reference/cors[CORS Configuration].
+
+|500
+|There was a FusionAuth internal error. A stack trace is provided and logged in the FusionAuth log files.
+|===
+
+== Register
+
+This API will start an OAuth 2 authorization code flow by building a valid request and then redirecting the browser to our `/oauth2/register` endpoint. This endpoint is nearly identical to the <<login>> endpoint with the exception that the target is registration instead of login. If the user is not logged in the user will be presented with the registration page and prompted for credentials before being redirected back to the <<callback>> endpoint.
+
+To use this API correctly the browser should be redirected to this route and it will respond with a redirect status code if successful.
+
+=== Request
+
+[.api-authentication]
+link:/docs/v1/tech/apis/authentication#no-authentication-required[icon:unlock[type=fas]] Start the registration flow
+
+[.endpoint]
+.URI
+--
+[method]#GET# [uri]#/app/register/{clientId}``?redirect_uri=\{redirectUri\}&state=\{state\}&scope=\{scope\}``#
+--
+
+==== Request Parameters
+
+[.api]
+[field]#clientId# [type]#[UUID]# [required]#Required#::
+The top-level client Id for your application.
+
+[field]#redirect_uri# [type]#[String]# [optional]#Optional#::
```

Comment:
Let the reader know that if a value is provided, it needs to be in the list of authorized URLs for the application. What happens if this isn't provided, and there are no authorized URLs defined in the app?

Reply:
I added a link in the prereqs section about the application configuration which covers this in more detail.